client/modules/transactions.js

```diff
@@ -0,0 +1,63 @@
+import {
+  defaultTransactions
+} from './defaults';
+
+import { actions as summaryActions } from './summary';
+
+const ADD_TRANSACTION = 'budgeting-sample-app/transaction/ADD_TRANSACTION';
+const DELETE_TRANSACTION = 'budgeting-sample-app/transaction/DELETE_TRANSACTION';
+
+export const actions = {
+  createTransaction: transaction => ({
+    type: ADD_TRANSACTION,
+    transaction
+  }),
+
+  deleteTransaction: id => ({
+    type: DELETE_TRANSACTION,
+    id
+  }),
+
+  addTransaction: transaction => (
+    (dispatch, getState) => {
+      const addedResult = dispatch(actions.createTransaction(transaction));
+      dispatch(summaryActions.requestSum(getState().transactions.transactions));
+      return addedResult;
+    }
+  )
+};
+
+/**
+ * Add a new transaction.
+ * This is a helper function for the transactions reducer
+ * @param {Object} state
+ * @param {Object} action
+ */
+function addTransactionToState(state, action) {
+  const { description, value } = action.transaction;
+  const newState = [...state, {
+    id: state.reduce((maxId, todo) => Math.max(todo.id, maxId), -1) + 1,
```

Comment:
can this reduce function be named somewhere outside addTransactionToState and called here? It might be easier to read then